### DISCUSSION TITLE
QualCoder 3.0

### DISCUSSION THREAD

**Discussion by ccbogel** [ccbogel/QualCoder#653] [Announcements]
# QualCoder 3.0 release information

I have only tested on Ubuntu 20.04 and Windows 10.
You must have VLC installed. 
Optionally, install ffmpeg for speech to text and waveform image creation. (ffmpeg tested on Ubuntu only).
There is an exe for Windows 10. 
There is an executable for Ubuntu 20.04. 

## Qt6
Moved the graphical framework from Qt5 to Qt6

## Manage files
- Added search text function when viewing a text file. Not done.
- Rename database file name entry function added.

## Code text
- Fixed uncaught IndexError exception when editing text positions. 
- Fixed AttributeError updating Tooltips in merge_codes
- Export coded text file to odt or html. Does not export the tooltips (future work to do).

## Reports
- Added Charting function.
- Pie, Bar, Sunburst, Heatmap and Treemap charts can be created and are displayed in the default web browser.
- Heatmaps are limited to 40 rows and 40 columns for nicer display. 

- Graph 
  - Added functions to show files and cases as text objects.
  - Added function to add extra free text objects and lines.
  - zoom +/- function.  
  - Future work will be to save and load these user edited graphs within the database.

## Special functions
- Added ability to merge another QualCoder project into the current project. Always do a back up of your project first.

## Errors fixed
 - File deletion error - SQL bindings mismatch - fixed.
 -  Uncaught IndexError exception when editing text positions from within code text - fixed.
 -  AttributeError updating Tooltips when merging codes- fixed.
  - Dark text on dark code colour in Code image fixed. Now shows white text. fixed.

## Read the Readme file for installation from source
  

<hr /><em>This discussion was created from the release <a href='https://github.com/ccbogel/QualCoder/releases/tag/3.0'>QualCoder 3.0</a>.</em>

**Comment by glocalglocal** [comment]
> Added search text function when viewing a text file

I don't see this anywhere. Do markdown files (.md) count as text files?

**Comment by ccbogel** [comment]
Yes. This is activated with the right-click menu.
I will fix that right-click menu issue and upload new binaries for th3 3.0 release.
regards CC
________________________________

**Comment by ccbogel** [comment]
Actually, no, I missed this one. to do for the next version.
________________________________

**Comment by glocalglocal** [comment]
Ideally, qc would open any text file as text. Or the users could specify in Settings the file extensions they want for text. There are too many potential and unpredictable extensions.
